abi.md

```diff
@@ -110,6 +110,127 @@ Passive bundles shall be lowered to Verilog packed structs.
 Reference types in ports shall be logically split out from aggregates and named
 as though "Aggregate Type Lowering" was used.
 
+## On Groups
+
+Optional groups are lowered to separate modules that are instantiated via
+SystemVerilog `bind`{.verilog} statements.  Each module that contains group
+instances will produce an additional module whose name is derived from the group
+and nested group instance.  Nested groups are appended.  E.g.,
+`<module-name>$g<group-name>` is the name of a generated module for its first
+group.
+
+For each optional group, one binding file will be produced with a name derived
+from the group and nested group instance.
+`bindings$c<circuit-name>$g<group-name>.sv`.  The instantiation name for the
```

Comment:
```suggestion
`bindings$c<circuit-name>$g<group-name>[$g<group-name>].sv`.  The instantiation name for the
```

Reply:
im not sure that nesting the group names in the file names is getting us anything?

Reply:
I'm actually not super keen on this, but there is a big benefit of doing it this way: this allows for the group names to conflict with each other. If the group names _aren't_ appended, then you need to unique these in some way. I.e., this allows:

```
circuit Foo:
  defgroup A:
    defgroup A:
```

Otherwise, all group names, including nested names, need to be unique. This may be fine.

Reply:
i feel like the unique name thing might be easier to implement 🤷‍♀️

Reply:
Going to just flatten this out. If group names, including nested group names, are unique in the module namespace, then we can do this. From the perspective of a user, they aren't thinking about nested groups, just this circuit has all these groups and I can choose any of them to bind in.